README.md

```diff
@@ -157,6 +157,106 @@ This is useful for long examples.
 
 [^back to top](#recog-a-recognition-framework)
 
+### Examples
+
+The following examples make use of `bin/recog_match`, a simple Ruby command line tool that uses Recog's fingerprinting data. Pre-processing is generally required before running Recog, i.e. extracting HTTP header values, etc. All fingerprint data can be found in `xml/*.xml`.
+
+### ftp_banners
+
+Fingerprint FTP servers based on the server's banner response after connecting:
+
+```
+# Example plaintext input
+echo -n '220 Microsoft FTP Service' | bin/recog_match xml/ftp_banners.xml -
+
+# Example command using nmap
+nmap -sV -script=banner -p 21 192.168.123.13 | grep '_banner' | cut -d: -f2- | bin/recog_match xml/ftp_banners.xml -
+
+# Example output
+MATCH: {"matched"=>"Microsoft FTP Server on Windows XP, 2003 or later without version", "service.vendor"=>"Microsoft", "service.product"=>"IIS", "service.family"=>"IIS", "service.cpe23"=>"cpe:/a:microsoft:iis:-", "os.vendor"=>"Microsoft", "os.family"=>"Windows", "os.product"=>"Windows", "os.cpe23"=>"cpe:/o:microsoft:windows:-", "host.name"=>"220", "service.protocol"=>"ftp", "fingerprint_db"=>"ftp.banner", "data"=>"220 Microsoft FTP Service"}
+```
+
+#### http_cookies
+
+Using the HTTP `Set-Cookie` header value to fingerprint an HTTP server:
+
+```
+# Example plaintext input
+echo 'laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | bin/recog_match xml/http_cookies.xml -
+
+# Example cURL command
+curl --silent -I http://localhost:9001 | grep -i '^Set-Cookie:' | cut -d: -f2- | bin/recog_match xml/http_cookies.xml -
```

Comment:
The `cut -d: -f2-` will end up stripping additional colon characters which may exist in the fingerprint data and could cause confusion for user's that might copy and paste from the examples. Could we modify these examples so that they only strip the first (header) colon character? (See other examples)

Reply:
Do you have replication steps for this? I originally spent the extra cycle to make sure it didn't strip additional colon characters - but maybe I've missed an edgecase 😄

```
$ echo 'Set-Cookie: foo=a:b:c:d' | cut -d: -f2- 
  foo=a:b:c:d
```

Verified working in the same way on mac and linux, as well as checking posix mode in bash

Reply:
I wish I would have shared the command I ran when I made that comment! Based on where I commented I think I reversed the issue. I believe I may have accidentally deleted the trailing `-` when testing the failed command after adding the header to the string.

```
$ echo 'laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | bin/recog_match xml/http_cookies.xml -
MATCH: {"matched"=>"Laravel PHP web application framework", "service.vendor"=>"Laravel", "service.product"=>"Laravel", "service.cpe23"=>"cpe:/a:laravel:laravel:-", "service.protocol"=>"http", "fingerprint_db"=>"http_header.cookie", "data"=>"laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax"}

$ echo 'Set-Cookie: laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | grep -i '^Set-Cookie:' | cut -d: -f2- | bin/recog_match xml/http_cookies.xml -
FAIL:  laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax


$ echo 'Set-Cookie: laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | grep -i '^Set-Cookie:' | cut -d: -f2-
 laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax

$ echo 'Set-Cookie: laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | grep -i '^Set-Cookie:' | cut -d: -f2
 laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16
```

Reply:
Turns out this confusion was caused by me having grep aliased to `grep --color=always`. 😅 

```
$ echo 'Set-Cookie: laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | grep -i '^Set-Cookie:' | cut -d: -f2- | hexdump -C
C
00000000  1b 5b 6d 1b 5b 4b 20 6c  61 72 61 76 65 6c 5f 73  |.[m.[K laravel_s|
00000000  1b 5b 6d 1b 5b 4b 20 6c  61 72 61 76 65 6c 5f 73  |.[m.[K laravel_s|
...

$ echo 'Set-Cookie: laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | grep -i --color=never '^Set-Cookie:' | cut -d: -f2- | hexdump -CC
00000000  20 6c 61 72 61 76 65 6c  5f 73 65 73 73 69 6f 6e  | laravel_session|
00000000  20 6c 61 72 61 76 65 6c  5f 73 65 73 73 69 6f 6e  | laravel_session|
...

$ echo 'Set-Cookie: laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax' | grep -i --color=never '^Set-Cookie:' | cut -d: -f2- | bin/recog_match xml/http_cookies.xml -
MATCH: {"matched"=>"Laravel PHP web application framework", "service.vendor"=>"Laravel", "service.product"=>"Laravel", "service.cpe23"=>"cpe:/a:laravel:laravel:-", "service.protocol"=>"http", "fingerprint_db"=>"http_header.cookie", "data"=>"laravel_session=eyJ...etc..etc...%3D; expires=Mon, 13-Mar-2023 16:48:58 GMT; Max-Age=7200; path=/; httponly; samesite=lax"}
```

Reply:
Thanks for taking a closer look!

Let me know if I should put up a PR to add the ` --color=never ` option to the docs, as other users might run into it too 😄

Reply:
I was planning on opening one for this today. ;)